src/xrpld/app/tx/detail/apply.cpp

```diff
@@ -113,12 +146,88 @@ apply(
     ApplyFlags flags,
     beast::Journal j)
 {
-    STAmountSO stAmountSO{view.rules().enabled(fixSTAmountCanonicalize)};
-    NumberSO stNumberSO{view.rules().enabled(fixUniversalNumber)};
+    return apply(app, view, [&]() mutable {
+        return preflight(app, view.rules(), tx, flags, j);
+    });
+}
 
-    auto pfresult = preflight(app, view.rules(), tx, flags, j);
-    auto pcresult = preclaim(pfresult, app, view);
-    return doApply(pcresult, app, view);
+std::pair<TER, bool>
+apply(
+    Application& app,
+    OpenView& view,
+    uint256 const& batchId,
+    STTx const& tx,
+    ApplyFlags flags,
+    beast::Journal j)
+{
+    return apply(app, view, [&]() mutable {
+        return preflight(app, view.rules(), batchId, tx, flags, j);
+    });
+}
+
+static bool
+applyBatchTransactions(
+    Application& app,
+    OpenView& batchView,
+    STTx const& txn,
```

Comment:
```suggestion
    STTx const& batchTxn,
```
Lots of transactions in this function